sdk/ai/azure-ai-projects/README.md

```diff
@@ -463,6 +553,81 @@ agent = await project_client.agents.create_agent(
 
 <!-- END SNIPPET -->
 
+#### Create Agent With Azure Function Call
+
+The agent can handle Azure Functions on the service side and return the result of the call. To use the function we need to create the `AzureFunctionTool`, which contains the input and output queues of azure function and the description of input parameters. Please note that in the prompt we are asking model to invoke queue when the specific question ("What would foo say?") is being asked. 
+
+<!-- SNIPPET sample_agents_azure_functions.create_agent_with_azure_function_tool -->
+
+```python
+azure_function_tool = AzureFunctionTool(
+    name="foo",
+    description="Get answers from the foo bot.",
+    parameters={
+        "type": "object",
+        "properties": {
+            "query": {"type": "string", "description": "The question to ask."},
+            "outputqueueuri": {"type": "string", "description": "The full output queue uri."},
+        },
+    },
+    input_queue=AzureFunctionStorageQueue(
+        queue_name="azure-function-foo-input",
+        storage_service_endpoint=storage_service_endpoint,
+    ),
+    output_queue=AzureFunctionStorageQueue(
+        queue_name="azure-function-tool-output",
+        storage_service_endpoint=storage_service_endpoint,
+    ),
+)
+
+agent = project_client.agents.create_agent(
+    model="gpt-4",
+    name="azure-function-agent-foo",
+    instructions=f"You are a helpful support agent. Use the provided function any time the prompt contains the string 'What would foo say?'. When you invoke the function, ALWAYS specify the output queue uri parameter as '{storage_service_endpoint}/azure-function-tool-output'. Always responds with \"Foo says\" and then the response from the tool.",
+    tools=azure_function_tool.definitions,
+)
+print(f"Created agent, agent ID: {agent.id}")
+```
+
+<!-- END SNIPPET -->
+
+#### Create Agent with OpenAPI specification
+
+OpenAPI specifications describe REST operations against a specific endpoint. Agents SDK can read an OpenAPI spec, create a function from it, and call that function against the REST endpoint without additional client-side execution.
+
+Here is an example creating an OpenAPI tool (using anonymous authentication):
+
```

Comment:
Do we need to add a note here about endpoints with api-key authentication? Should we say that's coming soon?

Reply:
I think we leave as is given the pending CELA for the working auth sample...the customers in private preview (who care most about auth) should have access to that sample soon.